src/popover.ts

```diff
@@ -179,7 +186,10 @@ export function apply() {
     const target = event.target;
     if (!(target instanceof Element)) return;
     const doc = target.ownerDocument;
-    let effectedPopover = closestComposed(event);
+    let effectedPopover = (event
+      .composedPath()
```

Comment:
IIRC `composedPath` does not work with closed shadow roots. It may be worth adding tests for both open and closed shadow roots for this.

Reply:
refactored `composedPath` to a custom `closest` function. didn't override the prototype `closest` to keep consumers' environment intact